Updated git-annex documentation

kousu opened the pull request.

kousu committed:
Give git-annex download instructions.

kousu committed:
Document how to upload via git-annex.

jcohenadad commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,40 @@
+git add yourusername git@github.com:yourusername/data-multi-subject.git
+git annex sync yourusername
+```
+2. Start a branch:
````
```suggestion
3. Start a branch:
```

jcohenadad replied:
(pushing all the numbers below 😉 )

kousu replied:
True. I'll fix that.

jcohenadad reviewed (commented):
awesome! just the minor issue with the numbering. thanks 🙏

kousu committed:
Document common git-annex issues.

I expect this section to expand over time. I think there's a lot of gotchas
in git-annex.

kousu commented:
@jcohenadad I added the section on common issues I promised earlier today too.

kousu commented:
@jcohenadad you'll have to merge this. I'm not a member of this repo yet.

jcohenadad commented:
hum we should probably fix CI before being able to merge anything

jcohenadad commented:
@kousu should we also add doc for maintainer of this repos? e.g., i was trying to clone this repos on compute canada (to start doing some processing), and ran into:
~~~
git clone https://github.com/spine-generic/data-multi-subject.git
cd data-multi-subject
git annex init
git-annex: getUserEntryForID: does not exist (no such user)
~~~

kousu commented:
> @kousu should we also add doc for maintainer of this repos?

My intention was that, yeah, this would cover that. In CONTRIBUTING.md there's a troubleshooting section. We can dump things we run into in there. I expect there to be many glitches.

kousu committed:
Cover the hidden gotcha that git-annex is often out of date

kousu commented:
I've added a note to the docs about checking your git-annex version.

datalad's docs on the matter just say https://www.datalad.org/get_datalad.html

> There is one additional dependency — git-annex — that is not installed automatically when using pip. Be sure to have a recent version of git-annex installed on your system.

I tried to give a bit more guidance.

kousu commented:
Also these docs should probably live in a more general wiki but can they just go here for now pretty please?

kousu committed:
Explain that we have to pin to git annex v8

https://github.com/spine-generic/data-multi-subject_DO-NOT-USE/issues/20#issuecomment-674639768

kousu committed:
Update README.md

kousu commented:
> hum we should probably fix CI before being able to merge anything

@jcohenadad #6

jcohenadad commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,102 @@
+7. Publish:
+```
+git annex sync --content amazon
+git annex sync yourusername
````
```suggestion
git annex sync yourusername  # if you cloned this repository and you are not working on a fork, you can omit 'yourusername'
```

kousu replied:
Depends: are we going to be working in branches or working in forks? You told me before you wanted to protect this dataset and so wanted to force everything to be done by forks.

It could go either way but I think we should pick one and write the instructions to that. If someone wants to do it another way they can ask around and figure it out.

jcohenadad replied:
i mostly wrote this for myself 😬  (i cloned this repos and didn't feel like forking it and re-downloading everything)

jcohenadad commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,102 @@
+git annex sync --content amazon
+git annex sync yourusername
+```
+8. Go to https://github.com/yourusername/data-multi-subject and click the Pull Request button.
````
@kousu i see these 3 PR options:

![image](https://user-images.githubusercontent.com/2482071/90458653-fcf3ae80-e0cc-11ea-8d9b-5375f9fc65fb.png)

should it be "jca/participants"?

kousu replied:
Yes. `git-annex` and github aren't really designed to work together I guess!

jcohenadad commented on README.md:
```diff
@@ -15,30 +15,30 @@ zip codes, dates of birth, facial features on structural scans.
+## Download
 
-If you are only planning on using the dataset for processing, you can download the latest version as a zip package:
+We are using a tool to manage large datasets alled `git-annex`. To download, dataset, you need to have `git` installed, and also [install `git-annex`](https://git-annex.branchable.com/install/) *at version 8*. Then run:
```
```suggestion
We are using a tool to manage large datasets called `git-annex`. To download, dataset, you need to have `git` installed, and also [install `git-annex`](https://git-annex.branchable.com/install/) *at version 8*. Then run:
```

kousu replied:
:+1:

jcohenadad commented on README.md:
```diff
@@ -15,30 +15,30 @@ zip codes, dates of birth, facial features on structural scans.
+## Download
 
-If you are only planning on using the dataset for processing, you can download the latest version as a zip package:
+We are using a tool to manage large datasets alled `git-annex`. To download, dataset, you need to have `git` installed, and also [install `git-annex`](https://git-annex.branchable.com/install/) *at version 8*. Then run:
```
```suggestion
We are using a tool to manage large datasets alled `git-annex`. To download this dataset, you need to have `git` installed, and also [install `git-annex`](https://git-annex.branchable.com/install/) *at version 8*. Then run:
```

jcohenadad commented on README.md:
````diff
@@ -15,30 +15,30 @@ zip codes, dates of birth, facial features on structural scans.
+You may **substitute** `git annex get` with more specific commands if you are only interested in certain subjects. For example:
+
+```
+get annex get sub-nwu01/ sub-nwu03/ sub-nwu04/ sub-oxfordFmrib04/ sub-tokyoSkyra*/
````
easier to type
```suggestion
get annex get sub-nwu01 sub-nwu03 sub-nwu04 sub-oxfordFmrib04 sub-tokyoSkyra*
```

kousu replied:
I read the `/`s as implying "folder". Also, they show up automatically if you use tab-completion.

jcohenadad replied:
^ i'm fine with that

jcohenadad commented on README.md:
```diff
@@ -15,30 +15,30 @@ zip codes, dates of birth, facial features on structural scans.
 
 The processing pipeline to analyze this dataset is available in the [spine-generic repository](https://github.com/sct-pipeline/spine-generic).
```
duplicate of section analysis below
```suggestion
```

kousu committed:
Typos

kousu committed:
Apply suggestions from code review

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>
